### ISSUE TITLE
ssl with certificates

### ISSUE THREAD

**Issue by goock** [timgit/pg-boss#89]
Hi,
My SSL connection requires certificate at client side therefore simple true/false does not work for me.
This is my ssl parameter:
```
      ssl: {
        rejectUnauthorized : false,
        ca   : fs.readFileSync("./client-certs/server-ca.pem").toString(),
        key  : fs.readFileSync("./client-certs/client-key.pem").toString(),
        cert : fs.readFileSync("./client-certs/client-cert.pem").toString()
      },
```
I modify the ssl parameter as follow:
```
      ssl: typeof poolConfig.ssl === 'boolean' ? !!poolConfig.ssl : poolConfig.ssl
```
in ```pg-boss/lib/db.js.```

Would be possible to modify the official repo?

Thanks,
Marcin

**Comment by asafh** [comment]
Hi @timgit, thanks for this.  
The above issue is resolved in runtime but is not reflected in the types.d.ts file.  

I wouldn't mind creating a pull request for this - are you ok with adding a dev-dependency on @types/pg and simply having the DatabaseOptions type be an the import of PoolConfig?

Cheers

**Comment by timgit** [comment]
dev dependency is fine. Thanks

**Comment by on2air** [comment]
hi @asafh / @timgit  - any progress on the type modification for ssl?  I'm hacking it with
 `ssl?: boolean|any`
 to get a quick fix, but hoping to avoid in future. thx

**Comment by asafh** [comment]
I forgot about it but now that've reminded me, I've created the PR.
@timgit looks ok?
Cheers

**Comment by timgit** [comment]
Yes. Thanks

**Comment by timgit** [comment]
This is merged and published in 4.3.3

**Comment by asafh** [comment]
Hi @timgit , I realize now the solution is slightly wrong. Since devDependencies of pg-boss would not be installed, a Typescript project using it would raise an issue when it reaches the import:
``` 
node_modules/pg-boss/types.d.ts(2,26): error TS7016: Could not find a declaration file for module 'pg'. '/home/node/some_app/node_modules/pg/lib/index.js' implicitly has an 'any' type.  
Try `npm install @types/pg` if it exists or add a new declaration (.d.ts) file containing `declare module 'pg';
````

I think the right approach is to change it to a normal dependency, is that ok or would you suggest otherwise?
[moved my comment from the PR to here.]

**Comment by timgit** [comment]
How about we punt to the any type for now as originally suggested?

```
ssl?: boolean|any
```

**Comment by asafh** [comment]
Another suggestion is to have @types/node as a normal dependency and use the tls module's ConnectionOptions (which is the type for the ssl parameter in PoolConfig) but this does have a similar need for a production dependency for a @types package.  

Let me know which option seems reasonable to you:
1. Leaving the current solution which means any Typescript package using pg-boss needs to explicitly include @types/pg in its package dev-depedency.  
2. Change the @types/pg dependency to production one and keep this most precise export PoolConfig from it @types/pg. 
3. Replace the @types/pg dev-dep for a smaller @types/node production dependency and use the tls package's ConnectionOptions for the ssl property (while manually copying the other properties from PoolConfig at @types/node  package like before). 

My vote goes to (1) since its leaves the burden for typescript users only while giving them the most precise typing info, but these options are all valid and naturally its your decision.

**Comment by timgit** [comment]
I'm currently planning on republishing this with the following definition.  It's not a perfect abstraction over the pg module, but it does provide the vast majority of in-use props.

```
  interface DatabaseOptions {
    application_name?: string;
    database?: string;
    user?: string;
    password?: string;
    host?: string;
    port?: number;
    schema?: string;
    ssl?: boolean | object;
    connectionString?: string;
    poolSize?: number;
    max?: number;
    db?: Db;
  }
```

**Comment by asafh** [comment]
Sounds good and thanks for maintaining this project!

**Comment by timgit** [comment]
Included in 4.3.4
